### PR TITLE
Tickets/dm 11284

### PR DESCRIPTION
**Pull request by n8pease**

### CODE REVIEW

File: tests/test_repository.py [hunk 1]
```diff
@@ -840,6 +815,135 @@ def makeRegistry(location, name):
...
+                               outputs=repoBDir)
+        # initializing the butler, but only specifiying the root of 'repoB' (the rest of the cfg will be
+        # loaded from its `repositoryCfg` file), but fully specify the input as required.
+        butler = dp.Butler(inputs={'root': self.repoADir,
+                                   'mapperArgs': {'foo': 1}},
+                           outputs=repoBDir)
+        self.assertEqual(butler._repos.inputs()[0].repo._mapper.kwargs, {'foo': 1})
+        # use 'repoB' as an input, and verify that 'repoA' is loaded, including the mapperArgs.
+        butler = dp.Butler(inputs=repoBDir)
+        self.assertEqual(butler._repos.inputs()[1].repo._mapper.kwargs, {'foo': 1})
```
Comment (andy-slac) [inline tests/test_repository.py]: I guess assumption here is that `inputs()[1]` refers to repoA and not repoB? Is that ordering checked by some other test?

File: tests/test_repository.py [hunk 2]
```diff
@@ -840,6 +815,135 @@ def makeRegistry(location, name):
...
+
+
+class TestOldButlerParentTagging(unittest.TestCase):
+    """A test to verify that when a parent is an old butler repo that it still gets loaded correctly,
+    including mapperArgs and tagging."""
```
Comment (andy-slac) [inline tests/test_repository.py]: This is the same comment as for `TestOldButlerParent`, do they test the same thing?

File: python/lsst/daf/persistence/test/testMapper.py [hunk 3]
```diff
@@ -21,11 +21,37 @@
...
+    def __init__(self, root, **kwargs):
+        self.root = root
+        self.storage = Storage.makeFromURI(self.root)
+
+    def map_foo(self, dataId, write):
+        python = TestObject
+        persistable = None
+        storage = 'PickleStorage'
+        fileName = 'filename'
+        for key, value in dataId.items():
```
Comment (andy-slac) [inline python/lsst/daf/persistence/test/testMapper.py]: Maybe `sorted(dataId.items())` to make sure that file name does not depend on the order of the items returned from `items()` (I think that order is not guaranteed to be identical even if content of dict is identical though maybe I'm confusing it with something else)?

File: python/lsst/daf/persistence/butler.py [hunk 4]
```diff
@@ -725,23 +727,29 @@ def _getOldButlerRepositoryCfg(repositoryArgs):
...
+            and is used as the cfgRoot (URI to the location of the cfg file).
 
         Returned
         --------
         (RepositoryCfg or None, bool)
             The RepositoryCfg, or None if one cannot be found, and True if the RepositoryCfg was created by
             reading an Old Butler repository, or False if it is a New Butler Repository.
         """
+        if not isinstance(repositoryArgs, RepositoryArgs):
+            repositoryArgs = RepositoryArgs(cfgRoot=repositoryArgs, mode='r')
```
Comment (n8pease) [inline python/lsst/daf/persistence/butler.py]: nice catch. copy-paste error on the docs. it should be `repositoryArgs : RepositoryArgs or string`, will fix.

File: tests/test_repository.py (hunk 1 above)
Comment (n8pease) [inline tests/test_repository.py]: yes, inputs & parents and repo order is covered by other tests.

File: python/lsst/daf/persistence/test/testMapper.py (hunk 3 above)
Comment (n8pease) [inline python/lsst/daf/persistence/test/testMapper.py]: yeah, good catch.

**Pull request merged**